src/tr/tr.rs

```diff
@@ -73,21 +84,21 @@ fn tr<'a>(set1: ExpandSet<'a>, mut set2: ExpandSet<'a>) {
         map.insert(i as usize, s2_prev);
     }
 
-    let mut reader = BufReader::new(stdin());
-    let mut writer = BufWriter::new(stdout);
 
-    while let Ok(length) = reader.read_to_string(&mut buf) {
-        if length == 0 { break }
+    // read_line would automatically lock stdin,
+    // but then release the lock once the line is read,
+    // resulting in locking and unlocking of stdin for every line read.
+    // Lock it only once here instead
+    stdin.lock();
```

Comment:
I guess `Stdin::lock` does not work as you expected.

`Stdin::lock()` returns a `StdinLock` instance and locks the handle. As the lifetime of the instance ends, the lock gets released.

IMHO, to lock stdin we should:

```rust
let mut handle = stdin.lock()
// same as `Stdin::read_line`
handle.read_line(&mut buf)
```

PS: You could just reply to my comments.